docs/academy/module5.md

```diff
@@ -0,0 +1,80 @@
+
+## The LwM2M way of updating firmware
+
+Remote firmware updates knows many names, such as Firmware Update OTA (FUOTA) or Device Firmware Upgrade (DFU). Device vendors can develop their own custom FOTA mechanisms, however, these implementations are specific to the vendor or platform and cannot be easily ported to other systems. The problem of vendor compatibility can be resolved by adopting the LwM2M standard, which is hardware agnostic and can be implemented on any hardware platform.
+
+The LwM2M firmware update process consists of four steps: 
+
+1. The LwM2M Client is **triggered** to initiate the firmware update process
+1. The Client **downloads** the firmware and reports to the LwM2M Server when the download is finished
+1. The Client performs the **firmware update** after validating the integrity and authenticity of the new firmware which is done through a process called **secure boot**.
+1. The Client attempts to **run the new firmware** and reports the status to the Server. If the update is successful, the device starts running the new firmware, if not, the device performs a **rollback** to the earlier firmware version.
+
+This procedure is defined in detail in the LwM2M specifications (see [LwM2M specifications](https://www.openmobilealliance.org/release/LightweightM2M/V1_1_1-20190617-A/HTML-Version/OMA-TS-LightweightM2M_Core-V1_1_1-20190617-A.html#13-6-1-0-E61-Firmware-Update-State-Machine)). To add firmware update capabilities to a device, the [Firmware Update Object /5](http://devtoolkit.openmobilealliance.org/OEditor/LWMOView?url=https%3A%2F%2Fraw.githubusercontent.com%2FOpenMobileAlliance%2Flwm2m-registry%2Fprod%2F5.xml) needs to be implemented containing all essential functionalities for conducting the update and reporting the status. Several LwM2M Clients natively support this Firmware Update Object, including [AVSystem’s Anjay LwM2M Client](https://www.avsystem.com/products/anjay/) and [Zephyr’s LwM2M Client](https://docs.zephyrproject.org/latest/samples/net/lwm2m_client/README.html).
+
+![FOTA in progress](images/module5_in-progress-basic.png)
+*<p style="text-align: center;">Updating a device’s firmware using Coiote IoT Device Management</p>*
+
+## Download a new firmware
+The LwM2M standard defines two methods to perform a firmware update, allowing either the Client or the Server to decide on the best moment to start the download of the new firmware. 
+
+**PULL method**
+In the PULL scenario, the LwM2M Server provides the device with the address of the server containing the firmware file known as the **Package URI**. The device subsequently downloads the firmware from the so-called firmware repository at the earliest available opportunity.
+
+**PUSH method**
```

Comment:
I think it shouldn't look like this. I mean the bold PULL and PUSH and the sentence after this.
![image](https://github.com/AVSystem/developer-zone/assets/61635265/50f3bcd6-b442-49a6-94c2-db232e72b527)